src/utils/path.js

```diff
@@ -0,0 +1,34 @@
+const removeMultiSlashes = (path) => path.replace(/\/+/g, '/');
+const stripTrailingSlashes = (path) => path.replace(/\/+$/, '');
+const addLeadingSlash = (path) => path.charAt(0) === '/' ? path : '/' + path;
+const stripBasePath = (path, basePath) => basePath !== '/' && path.indexOf(basePath) === 0 ? path.substr(basePath.length) : path;
+
+export const parsePath = (path, options) => {
+  let pathname = stripBasePath(path, options.basePath);
+  let hash = '';
+  let anchor = null;
+
+  const hashIndex = pathname.indexOf('#');
+  if (hashIndex !== -1) {
```

Comment:
Thanks @wereHamster, I looked at this the wrong way: `-1` is not an integer, it's just how this fact is being represented. Schön hämmer dröber nodänkt ;)